File: 2892-check-if-array-is-good/check-if-array-is-good.cpp
class Solution {
public:
    bool isGood(vector<int>& nums) {
        int n = nums.size() - 1;
        unordered_set<int> seen;
        bool dup = false;

        for (int num : nums) {
            if (num > n) {
                return false;
            }
            if (seen.count(num)) {
                if (num < n || dup) {
                    return false;
                }
                dup = true;
                continue;
            }
            seen.insert(num);
        }
        return true;
    }
};

File: 2892-check-if-array-is-good/check-if-array-is-good.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isGood = function (nums) {
    nums.sort((a, b) => a - b)
    const n = nums.length - 1
    for (let i = 0; i < n; i++) {
        if (nums[i] !== i + 1) {
            return false
        }
    }
    return nums[n] === n
};